### PR TITLE
Enable fp16 on CPU

### PR DESCRIPTION
**Pull request by muellerzr**
# What does this PR do?

This PR enables users to allow ops that can be done on CPU in FP16 to be done, without blocking them.

(Note: how well they work is TBD, at least the examples tests pass which have a threshold, but edge with caution)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@amyeroberts @ydshieh

### CODE REVIEW

File: src/transformers/training_args.py [hunk 1]
```diff
@@ -1618,7 +1618,7 @@ def __post_init__(self):
         if (
             self.framework == "pt"
             and is_torch_available()
-            and (self.device.type == "cpu" and (version.parse(get_torch_version()) < version.parse("2.3.0")))
+            and (self.device.type == "cpu" and is_torch_greater_or_equal_than_2_3)
```
Comment (amyeroberts) [inline src/transformers/training_args.py:1621]: I think? 

```suggestion
            and (self.device.type == "cpu" and not is_torch_greater_or_equal_than_2_3)
```

**Review by amyeroberts — COMMENTED** [review]
Thanks! 

Just a comment on a logic check

File: src/transformers/training_args.py (hunk 1 above)
Comment (muellerzr) [reply src/transformers/training_args.py]: Yep! (sorry had computer in front of me and pushed right as you caught it 😓 )

**Review by amyeroberts — APPROVED** [review]
Thanks!

**Pull request merged**